internal/utils/k8s/utils_test.go

```diff
@@ -109,21 +112,18 @@ func TestMergePodAnnotations(t *testing.T) {
 		{
 			name: "should preserve existing checksum annotations",
 			existing: map[string]string{
-				"checksum/1": "1",
-				"checksum/2": "2",
-				"checksum/3": "3",
-				"unrelated":  "foo",
+				"checksum/config":   "1",
+				"checksum/config-a": "2",
+				"checksum/Config":   "3",
+				"unrelated":         "foo",
 			},
 			desired: map[string]string{
-				"checksum/2": "b",
-				"checksum/3": "3",
-				"checksum/4": "4",
+				"checksum/cOnfig": "6",
```

Comment:
previously this test case also checked that no existing entries in the desired map would be overridden...please add it back